src/components/HoldPages/EDDRequestForm.tsx

```diff
@@ -0,0 +1,282 @@
+import {
+  Form,
+  FormField,
+  ButtonGroup,
+  Button,
+  TextInput,
+  FormRow,
+  Box,
+  Heading,
+  Text,
+} from "@nypl/design-system-react-components"
+import { useState, createRef, useReducer, type SyntheticEvent } from "react"
+
+import { BASE_URL } from "../../config/constants"
+import ExternalLink from "../Links/ExternalLink/ExternalLink"
+
+import { CopyrightRestrictionsBanner } from "./CopyrightRestrictionsBanner"
+import { eddFormReducer } from "../../reducers/eddFormReducer"
+import {
+  initialEDDFormState,
+  getUpdatedInvalidFields,
+  validateEDDForm,
+  initialEDDInvalidFields,
+  isInvalidField,
+} from "../../utils/holdPageUtils"
+import type { EDDRequestParams, EDDPageStatus } from "../../types/holdPageTypes"
+
+interface EDDRequestFormProps {
+  handleSubmit: (eddParams: EDDRequestParams) => void
+  setPageStatus: (status: EDDPageStatus) => void
+  holdId: string
+  patronId: string
+  source: string
+}
+
+/**
+ * The EDDRequestForm renders the form for placing a electronic delivery hold on an item.
+ */
+const EDDRequestForm = ({
+  handleSubmit,
+  setPageStatus,
+  holdId,
+  patronId,
+  source,
+}: EDDRequestFormProps) => {
+  const [eddFormState, dispatch] = useReducer(eddFormReducer, {
+    ...initialEDDFormState,
+    patronId,
+    source,
+  })
+
+  // Set the invalid fields as an array in state to keep track of the first invalid field for focus on submit
+  const [invalidFields, setInvalidFields] = useState(initialEDDInvalidFields)
+
+  // Create refs for fields that require validation to focus on the first invalid field on submit
+  const [validatedInputRefs] = useState(
+    invalidFields.reduce((acc, field) => {
+      return { ...acc, [field.key]: createRef() }
+    }, {})
+  )
+
+  // Separate validation from input change to validate on tabbing (blur) and to prevent validation on every keystroke on fields other than email
+  const validateField = (e: SyntheticEvent) => {
+    e.preventDefault()
+    const target = e.target as HTMLInputElement
+
+    setInvalidFields((prevInvalidFields) =>
+      getUpdatedInvalidFields(target.name, target.value, prevInvalidFields)
+    )
+  }
+
+  const handleInputChange = (e: SyntheticEvent) => {
+    e.preventDefault()
+    const target = e.target as HTMLInputElement
+
+    dispatch({
```

Comment:
I think this reducer is overkill for the single kind of input we're handling here.
I'd love to see it removed and instead just using
`const [eddFormState, setEddFormstate] = useState(initialFormState)`
and then calling `setEddFormState` instead of a dispatch to the reducer.

Reply:
Good call! replace reducer with useState